src/mlpack/tests/distribution_test.cpp

```diff
@@ -442,6 +442,97 @@ BOOST_AUTO_TEST_CASE(GammaDistributionTrainTest)
   BOOST_REQUIRE_EQUAL(gDist.Dimensionality(), d2);
 }
 
+BOOST_AUTO_TEST_CASE(GammaDistributionTrainWithProbabilitiesTest)
+{
+  double alphaReal = 5.4;
+  double betaReal = 6.7;
+
+  // Create a gamma distribution random generator
+  std::default_random_engine generator;
+  std::gamma_distribution<double> dist(alphaReal, betaReal);
+
+  size_t N = 500;
+  size_t d = 2;
+  arma::mat rdata(d, N);
+
+  for(size_t j = 0; j < d; j++)
+    for(size_t i = 0; i < N; i++)
+      rdata(j, i) = dist(generator);
+
+  // create a uniform distribution random generator
+  std::uniform_real_distribution<double> prob(0, 1);
+  arma::vec probabilities(N);
+
+  for(size_t i = 0; i < N; i++)
+    probabilities(i) = prob(generator);
+
+  // fit results with probabilities and data
+  GammaDistribution gDist;
+  gDist.Train(rdata, probabilities);
+
+  // fit results with only data
+  GammaDistribution gDist2;
+  gDist2.Train(rdata);
+
+  BOOST_REQUIRE_CLOSE(gDist2.Alpha(0), gDist.Alpha(0), 10);
+  BOOST_REQUIRE_CLOSE(gDist2.Beta(0), gDist.Beta(0), 10);
+
+  BOOST_REQUIRE_CLOSE(alphaReal, gDist.Alpha(0), 10);
+  BOOST_REQUIRE_CLOSE(betaReal, gDist.Beta(0), 10);
```

Comment:
Ok, I think that we were getting confused because we were talking about probabilities for different tests. :)  I looked through the most recent code, and I think the tolerances are fine now.  When I was saying that the tolerance should be small like 1e-5, I was only referencing the test with probabilities all equal to one.  Thanks for the updates.